src/index.js

```diff
@@ -2,21 +2,27 @@ const core = require('@actions/core');
 const github = require('@actions/github');
 
 async function main() {
-  const { eventName, sha, ref, repo: { owner, repo } } = github.context;
-  const branch = ref.slice(11);
-  console.log({ eventName, sha, branch, owner, repo });
+  const { eventName, sha, ref, repo: { owner, repo }, payload } = github.context;
+  let branch = ref.slice(11);
+  let headSha = sha;
+  if (payload.pull_request) {
+    branch = payload.pull_request.head.ref;
```

Comment:
Do you need to do `ref.slice(11)` here too so that it read the correct branch name instead of the ref?

Reply:
From the usage I've done so far, the `payload.pull_request.head.ref` is already well named and doesn't need to be "sliced"

Reply:
Here's an example of payload I had
<img width="374" alt="Screenshot 2020-02-12 at 09 58 45" src="https://user-images.githubusercontent.com/1374161/74324024-4d84fa80-4d7e-11ea-98a1-9f94db2d5b19.png">